staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -1005,6 +1015,24 @@ func (alloc *allocator) allocateDevice(r deviceIndices, device deviceWithID, mus
 	}, nil
 }
 
+func allTaintsTolerated(device *draapi.BasicDevice, request requestAccessor) bool {
```

Comment:
This method is called from the `allocator.allocateDevice`.

https://github.com/kubernetes/kubernetes/blob/a0b3bbd1e3ea4ddfb4b3555a385a27f750217b33/staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go#L962-L968

It seems to take both `NoSchedule` and `NoExecute` taint/tolerations into consideration when allocating. I'm not actually sure, but, should we focus on `NoSchedule` taint only when allocating?

Reply:
NoExecute is the stronger one of the two effects and implies NoSchedule.

Suppose the scheduler ignores NoExecute and schedules a point using a device which is tainted with that effect. As soon as the pod is scheduled, the device eviction controller kicks in and deletes the pod because of that NoExecute.

Reply:
Sorry, node taint/toleration scheduler plugin takes both taint effects into considertion when scheduling 🙇 

https://github.com/kubernetes/kubernetes/blob/a0b3bbd1e3ea4ddfb4b3555a385a27f750217b33/pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go#L93-L99

https://github.com/kubernetes/kubernetes/blob/a0b3bbd1e3ea4ddfb4b3555a385a27f750217b33/pkg/scheduler/framework/plugins/helper/taint.go#L21-L28